Client/Frontend/Settings/NTPTableViewController.swift

```diff
@@ -9,49 +9,99 @@ import Shared
 import BraveRewards
 
 class NTPTableViewController: TableViewController {
-    let sponsoredRow = Row.boolRow(title: Strings.newTabPageSettingsSponsoredImages, option: Preferences.NewTabPage.backgroundSponsoredImages)
+    enum BackgroundImageType: RepresentableOptionType {
+        
+        case defaultImages
+        case sponsored
+        case superReferrer(String)
+        
+        var key: String {
+            displayString
+        }
+        
+        public var displayString: String {
+            switch self {
+            case .defaultImages: return "(\(Strings.NTP.settingsDefaultImagesOnly))"
+            case .sponsored: return Strings.NTP.settingsSponsoredImagesSelection
+            case .superReferrer(let referrer): return referrer
+            }
+        }
+    }
+    
+    init() {
+        super.init(style: .grouped)
+    }
+    
+    @available(*, unavailable)
+    required init?(coder aDecoder: NSCoder) {
+        fatalError()
+    }
     
     override func viewDidLoad() {
         super.viewDidLoad()
         
         // Hides unnecessary empty rows
         tableView.tableFooterView = UIView()
         
-        navigationItem.title = Strings.newTabPageSettingsTitle
+        navigationItem.title = Strings.NTP.settingsTitle
         tableView.accessibilityIdentifier = "NewTabPageSettings.tableView"
         dataSource.sections = [section]
-        
-        // Sponsored switch is only enabled if the background images is enabled also.
-        self.sponsoredSwitch?.isEnabled = Preferences.NewTabPage.backgroundImages.value
     }
     
-    private lazy var section: Section = {
-        var rows = [
-            Row.boolRow(title: Strings.newTabPageSettingsBackgroundImages, option: Preferences.NewTabPage.backgroundImages, onValueChange: {
-                newValue in
-                // Since overriding, does not auto-adjust this setting.
-                Preferences.NewTabPage.backgroundImages.value = newValue
-                
-                // If turning off normal background images, turn of sponsored images as well.
-                Preferences.NewTabPage.backgroundSponsoredImages.value = newValue
-                
-                // Update sponsored switch to both on/off to follow background images.
-                self.sponsoredSwitch?.isOn = newValue
-                
-                // Need to update every time.
-                self.sponsoredSwitch?.isEnabled = newValue
-            })
-        ]
+    private func selectedItem() -> BackgroundImageType {
+        if let referrer = Preferences.NewTabPage.selectedCustomTheme.value {
+            return .superReferrer(referrer)
+        }
         
-        if BraveAds.isCurrentLocaleSupported() {
```

Comment:
Are you sure we want to drop this logic? To not show 'sponsored images' toggle for non-eligible locales